source/utilities.cc

```diff
@@ -2167,17 +2167,10 @@ namespace aspect
       if (this->get_time() - first_data_file_model_time >= 0.0)
         {
           Point<dim> internal_position = position;
+          const std::array<double,dim> natural_position = this->get_geometry_model().cartesian_to_natural_coordinates(position);
 
-          if (dynamic_cast<const GeometryModel::SphericalShell<dim>*> (&this->get_geometry_model()) != nullptr
-              || dynamic_cast<const GeometryModel::Chunk<dim>*> (&this->get_geometry_model()) != nullptr
-              || dynamic_cast<const GeometryModel::Sphere<dim>*> (&this->get_geometry_model()) != nullptr)
-            {
-              const std::array<double,dim> spherical_position =
-                Utilities::Coordinates::cartesian_to_spherical_coordinates(position);
-
-              for (unsigned int i = 0; i < dim; i++)
-                internal_position[i] = spherical_position[i];
-            }
+          for (unsigned int i = 0; i < dim; i++)
+            internal_position[i] = natural_position[i];
```

Comment:
This looks easier :-)

You initialize `internal_position` above, but then overwrite it here. Can you just move the declaration of the variable right above its use here, and omit the initializer?

Reply:
Yes!